src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java

```diff
@@ -81,15 +87,42 @@ public Tenant save(UUID consortiumId, Tenant tenantDto) {
     }
 
     checkTenantNotExistsAndConsortiumExistsOrThrow(consortiumId, tenantDto.getId());
-    Tenant savedTenant = saveTenant(consortiumId, tenantDto);
+    TenantEntity savedTenantEntity = saveTenantEntity(consortiumId, tenantDto);
+    var savedTenant = converter.convert(savedTenantEntity, Tenant.class);
 
     runInFolioContext(createFolioExecutionContextForTenant(tenantDto.getId(), currentTenantContext, folioMetadata),
-      () -> configurationClient.saveConfiguration(createConsortiaConfigurationBody(centralTenantId)));
+        () -> {
+          configurationClient.saveConfiguration(createConsortiaConfigurationBody(centralTenantId));
+          createPrimaryUserAffiliationsAsync(consortiumId, toEntity(consortiumId, tenantDto), tenantDto);
+        });
     log.info("save:: saved consortia configuration with centralTenantId={} by tenantId={} context", centralTenantId, tenantDto.getId());
 
     return savedTenant;
   }
 
+  public CompletableFuture<Void> createPrimaryUserAffiliationsAsync(UUID consortiumId, TenantEntity consortiaTenant, Tenant tenantDto) {
+    return CompletableFuture
+      .runAsync(() -> {
+        log.info("Start creating user primary affiliation for tenant {}", tenantDto.getId());
+        var users = usersClient.getUserCollection("", 0, Integer.MAX_VALUE);
+        users.getUsers()
+          .forEach(user -> {
+            var consortiaUserTenant = userTenantRepository.findByUserIdAndTenantId(UUID.fromString(user.getId()), tenantDto.getId())
+              .orElse(null);
+            if (consortiaUserTenant != null && consortiaUserTenant.getIsPrimary()) {
+              log.debug("Primary affiliation already exists for tenant/user: {}/{}", tenantDto.getId(), user.getUsername());
+            } else {
+              userTenantService.createPrimaryUserTenantAffiliation(consortiumId, consortiaTenant, user.getId(), user.getUsername());
+            }
+          });
+      })
+      .thenAccept(v -> log.info("Successfully created primary affiliations for tenant {}", tenantDto.getId()))
+      .exceptionally(t -> {
+        log.error("Failed to create primary affiliations for new tenant", t);
+        return null;
+      });
```

Comment:
We also need to know was operation of creating primary affiliation was successful or not, for this we need to call mod-users to understand how many users exist and compare how many was processed.
For this I propose to use alredy existed table consortia_configuration with adding 2 new columns: before, after.
If counts are equal we can confirm that primary affiliations for all users were created. If they are not equals - we can re-run with force=true flag

Reply:
Also we have endpoint to get consortia configuration, so can use it to understand that operation was success if count matches

Reply:
consortia configuration should not be used to store/retrieve this info as this endpoint is public

Reply:
not sure we can know that primary aff job is completed/failed without logs as it requires mod-users schema, so probably no sense to save count somewhere

Reply:
It is just to make us save from issues that primary affiliation was not created for some existed users. Regarding concern that consortia-configuration is public - agree, we can add prefix /full to existing configuration endpoint and retrun all configuration protected by permission or add new field to internal tenant entity.
Also if we use dry-run this counts should be updated, that also adding complexity.
@grabsefx do you have some preferences do we need to store these count somewhere?

Reply:
We decided with @grabsefx to not introduce status flag now(it can be separate story), because lot of things need to clarify firstly, what are desired statuses possible, do we need to update status if some synchronus action failed etc. Instead of we decided to add more intensive logging with counts how may users processes, skipped during force run etc.